pkgs/development/python-modules/dnf4/wrapper.nix

```
@@ -0,0 +1,53 @@
+{ lib
+, wrapPython
+, python
+, stdenv
+, dnf4-unwrapped
+, dnf-plugins-core
+, plugins ? [ dnf-plugins-core ]
+}:
+let
+  pluginPaths = map (p: "${p}/${python.sitePackages}/dnf-plugins") plugins;
+in
+
+stdenv.mkDerivation {
+  name = "dnf4-${dnf4-unwrapped.version}";
+
+  outputs = [ "out" "man" "py" ];
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [
+    wrapPython
+  ];
+
+  propagatedBuildInputs = [
+    dnf4-unwrapped
+  ] ++ plugins;
+
+  makeWrapperArgs = [
+    (lib.optionalString (plugins != [ ])
+      ''--add-flags "--setopt=pluginpath=${lib.concatStringsSep "," pluginPaths}"'')
+  ];
+
+  installPhase = ''
+    mkdir -p $out
+    cp -R ${dnf4-unwrapped}/* $out/
+    mkdir -p $py
+    cp -R ${dnf4-unwrapped.py}/* $py/
+    mkdir -p $man
+    cp -R ${dnf4-unwrapped.man}/* $man/
```

```suggestion
    mkdir -p $out $py $man
    cp -R ${dnf4-unwrapped}/* $out/
    cp -R ${dnf4-unwrapped.py}/* $py/
    cp -R ${dnf4-unwrapped.man}/* $man/
```
also we should symlink instead

---

I couldn't get this running with symlinks, it keeps failing with permission errors:

```
post-installation fixup
chmod: changing permissions of '/nix/store/9mvni7wqni9q6nv5cff32w9njz679xl8-dnf4-4.18.2': Operation not permitted
chmod: changing permissions of '/nix/store/9mvni7wqni9q6nv5cff32w9njz679xl8-dnf4-4.18.2/lib': Operation not permitted
chmod: changing permissions of '/nix/store/9mvni7wqni9q6nv5cff32w9njz679xl8-dnf4-4.18.2/lib/systemd': Operation not permitted
chmod: changing permissions of '/nix/store/9mvni7wqni9q6nv5cff32w9njz679xl8-dnf4-4.18.2/lib/systemd/system': Operation not permitted
....
```

---

then we keep the cp for now unless someone else finds a good solution